python/desc/skycatalogs/objects/base_object.py

```diff
@@ -475,6 +401,170 @@ def get_LSST_fluxes(self, cache=True, as_dict=True, mjd=None):
         else:
             return list(fluxes.values())
 
+class SNObject(BaseObject):
```

Comment:
I have no objection to moving the `BaseObject` subclasses into their own files and moving out `CatalogContext`. What about the function `load_lsst_bandpasses` ?  It is also only loosely coupled to `BaseObject` and is similar to `CatalogContext` in that both define globals.
I'm inclined to leave `BaseObject` and `ObjectCollection` together since these are the two that need to be subclassed when adding new kinds of objects.

Reply:
`BaseObject` and `ObjectCollection` could be in the same file. Chris has suggested that the functionality in `load_lsst_bandpasses` should be in the rubin_sim package.   For now, it probably should be in a separate module so that it can be imported from different contexts (unless it's only used in one place, in which case, it can probably just live in that module.)

Reply:
About get_LSST_flux - the specific exclusion of object type 'sn' is not necessary since it's already the case that the fluxes will never be among the native attributes for them.

I'm not sure what to do about 

>  if cache and mjd is None:
>             setattr(self, att, val)

Here `mjd is None` is being used as a clumsy proxy for a static source. I don't know that source type is better since at some point there will be variable as well as static stars.

Reply:
One option would be to make the default implementation apply to static sources:
```
    def get_LSST_flux(self, band, sed=None, cache=True, mjd=None):
        # default implementation for static sources.
        if not band in LSST_BANDS:
            return None

        att = f'lsst_flux_{band}'
        # Check if it's already an attribute
        val = getattr(self, att, None)
        if val is not None:
            return val

        val = self.get_flux(lsst_bandpasses[band], sed=sed, mjd=mjd)

        if cache:
            setattr(self, att, val)
        return val
```
and override in the `SNObject` subclass with:
```
    def get_LSST_flux(self, band, sed=None, cache=True, mjd=None):
        if band not in LSST_BANDS:
            return None

        sed, _ = self._get_sed(mjd=mjd)
        return sed.calculateFlux(mjd)
```

Reply:
I've separated out the subclasses of `BaseObject` and `CatalogContext`. It involved some other changes to avoid circular dependencies - probably just as well. 
The bandpasses are only used in `BaseObject` so I left `load_lsst_bandpasses` alone.

Concerning `get_LSST_flux` I can do as you suggest for now.  What happens when we support variable stars?  Do we treat them all as variable (with no caching) even when they're not? Or override in StarObject as well with something that delves a little further to see if the star is variable or not?

Reply:
For adding variable stars, my initial inclination would be to split stars into at least two subclasses:  a static star class, which would be the current `StarObject` and a variable star subclass (or even subclasses) which would override `get_LSST_flux`.